src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/GeneratedFile.cs

```diff
@@ -0,0 +1,65 @@
+using NexusMods.Abstractions.MnemonicDB.Attributes;
+using NexusMods.Abstractions.MnemonicDB.Attributes.Extensions;
+using NexusMods.MnemonicDB.Abstractions;
+using File = NexusMods.Abstractions.Loadouts.Files.File;
+
+namespace NexusMods.Abstractions.Loadouts.Synchronizers;
+
+public static class GeneratedFile
+{
+    private const string Namespace = "NexusMods.Abstractions.Loadouts.Files.GeneratedFile";
+    
+    /// <summary>
+    /// The generator unique ID for the file, this specifies the class that is used to generate
+    /// the contents of this file. 
+    /// </summary>
+    public static readonly ClassAttribute<IFileGenerator> Generator = new("NexusMods.Abstractions.Loadouts.Files.GeneratedFile", nameof(Generator)) { IsIndexed = true };
+
+    public class Model(ITransaction tx) : File.Model(tx)
+    {
+        /// <summary>
+        /// Get an instance of the generator for this file.
+        /// </summary>
+        public IFileGenerator Generator
+        {
+            get => GeneratedFile.Generator.GetInstance(this);
+        }
+        
+        /// <summary>
+        /// Set the generator for this file.
+        /// </summary>
+        public void SetGenerator<TType>()
+        where TType : IFileGenerator
+        {
+            GeneratedFile.Generator.Add(this, TType.Guid);
+        }
+    }
+}
+
+/// <summary>
+/// Extensions for the <see cref="File.Model"/> class.
+/// </summary>
+public static class FileModelExtensions
+{
+    /// <summary>
+    /// If this file is a generated file, this will return true and cast the generated file to the out parameter.
+    /// </summary>
+    public static bool IsGeneratedFile(this File.Model model, out GeneratedFile.Model generatedFile)
+    {
+        generatedFile = model.Remap<GeneratedFile.Model>();
+        return model.Contains(GeneratedFile.Generator);
+    }
+
+    /// <summary>
+    /// Returns true if the file is a generated file of the specified type.
+    /// </summary>
+    public static bool IsGeneratedFile<TType>(this File.Model model)
+    where TType : IFileGenerator =>
+        model.TryGet(GeneratedFile.Generator, out var val) && val == TType.Guid;
+    
+    /// <summary>
+    /// Returns true if the file is a generated file.
+    /// </summary>
+    public static bool IsGeneratedFile(this File.Model model) =>
+        model.Contains(GeneratedFile.Generator);
+}
```

Comment:
This should go into a separate file. Similar to the `File.IsStoredFile` method, `IsGeneratedFile` should use the try-get pattern, see the comment on `File.IsStoredFile` for details. Also, please be consistent with these remappings. These methods are in an extension class while the `File.IsStoredFile` method is a class method on `File` itself. Do one or the other, not both.

Reply:
We can put it in separate files, but what's the reasoning for this? I like to keep these methods close to the things they are talking about/used with. Since attributes are all on a static class it makes sense to keep the remappings on the same classes as well. I ended up moving these into the same static class as GeneratedFile, but we can move them if we come up with a better idea. 

GeneratedFiles don't exist in the same project (due to dependency issues) as the other file models, so there's a bit of an issue there

Reply:
It's mostly about being consistent and predictable. Keeping this extension class in the same file is fine for now, but the other `IsSomething` and `TryGetAsSomething` methods should all be extension methods instead of member methods.